ci/check-documentation.py

```diff
@@ -79,6 +111,14 @@ def complain(message):
                     ' does not exist or is not a file')
                 continue
 
+            if md_path in index_child_md_links:
+                referenced_md_files.add(target_path)
+
```

Comment:
Please add a comment here to explain the check that the following section performs. (Like the comments that are above and below.)